skoolkit/basic.py

```diff
@@ -194,24 +254,152 @@ def _get_num_str(self, j):
             num_str = chr(self.snapshot[j]) + num_str
         return num_str[1:]
 
-    def _get_char(self, code):
-        if code == 94:
-            return '↑'
-        if code == 96:
-            return '£'
-        if code == 127:
-            return '©'
-        if 32 <= code <= 126:
-            return chr(code)
-        return '{{0x{:02X}}}'.format(code)
+class VariableLister:
+    def __init__(self):
+        self.text = TextReader()
 
-    def _get_token(self, code):
-        token = TOKENS[code]
-        if self.lspace and code >= 197 and token[0] >= 'A':
-            token = ' ' + token
-        self.lspace = True
-        if code < 168 or code == 203 or token[-1] in '#=>':
-            # RND, INKEY$, PI, THEN, '<=', '>=', '<>', 'OPEN #', 'CLOSE #'
-            return token
-        self.lspace = False
-        return token + ' '
+    def list_variables(self, snapshot):
+        lines = ['Variables:']
+        self.snapshot = snapshot
+        i = (snapshot[23627] + 256 * snapshot[23628]) or 23755
```

Comment:
Just assume that the variables start at VARS, and let any fallout happen if VARS is not set correctly (and get rid of _skip_basic_line()). I know BasicLister takes PROG to be 23755 if it's set to zero, but that's because I was initially testing it on snapshots created by bin2tap.py, which doesn't set system variables.

Later on I might add options to snapinfo.py for setting PROG and VARS explicitly.